src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -60,6 +60,177 @@ bool AletheProofPostprocessCallback::update(Node res,
+    // not generated.
+    //
+    //
+    // If F = false:
+    //
+    //                                    --------- IMPLIES_SIMPLIFY
+    //    T1                                 VP9
+    // --------- DUPLICATED_LITERALS      --------- EQUIV_1
+    //    VP8                                VP10
+    // -------------------------------------------- RESOLUTION
+    //          (cl (not (and F1 ... Fn)))*
+    //
+    // VP8: (cl (=> (and F1 ... Fn))) (cl (not (=> (and F1 ... Fn) false))
+    //      (not (and F1 ... Fn)))
+    // VP9: (cl (= (=> (and F1 ... Fn) false) (not (and F1 ... Fn))))
+    //
+    //
+    // Otherwise,
+    //                T1
+    //  ------------------------------ DUPLICATED_LITERALS
+    //   (cl (=> (and F1 ... Fn) F))**
+    //
+    //
+    // *  the corresponding proof node is (not (and F1 ... Fn))
+    // ** the corresponding proof node is (=> (and F1 ... Fn) F)
+    case PfRule::SCOPE:
+    {
+      bool success = true;
+
+      // Build vp1
+      std::vector<Node> negNode{d_cl};
+      std::vector<Node> sanitized_args;
+      for (Node arg : args)
+      {
+        negNode.push_back(arg.notNode());  // (not F1) ... (not Fn)
+        sanitized_args.push_back(d_anc.convert(arg));
+      }
+      negNode.push_back(children[0]);  // (cl (not F1) ... (not Fn) F)
+      Node vp1 = nm->mkNode(kind::SEXPR, negNode);
+      success &= addAletheStep(AletheRule::ANCHOR_SUBPROOF,
+                               vp1,
+                               vp1,
+                               children,
+                               sanitized_args,
+                               *cdp);
+
+      // Build vp2i
+      Node andNode;
+      if (args.size() != 1)
+      {
+        andNode = nm->mkNode(kind::AND, args);  // (and F1 ... Fn)
+      }
+      else
+      {
+        andNode = args[0];  // F1
+      }
+      std::vector<Node> premisesVP2 = {vp1};
+      std::vector<Node> notAnd = {d_cl, children[0]};  // cl F
+      Node vp2_i;
+      for (long unsigned int i = 0; i < args.size(); i++)
+      {
+        vp2_i = nm->mkNode(kind::SEXPR, d_cl, andNode.notNode(), args[i]);
+        success &=
+            addAletheStep(AletheRule::AND_POS, vp2_i, vp2_i, {}, {}, *cdp);
+        premisesVP2.push_back(vp2_i);
+        notAnd.push_back(andNode.notNode());  // cl F (not (and F1 ... Fn))^i
+      }
+
+      Node vp2a = nm->mkNode(kind::SEXPR, notAnd);
+      success &= addAletheStep(
+          AletheRule::RESOLUTION, vp2a, vp2a, premisesVP2, {}, *cdp);
+
+      notAnd.erase(notAnd.begin() + 1);  //(cl (not (and F1 ... Fn))^n)
+      notAnd.push_back(children[0]);     //(cl (not (and F1 ... Fn))^n F)
+      Node vp2b = nm->mkNode(kind::SEXPR, notAnd);
+      success &=
+          addAletheStep(AletheRule::REORDER, vp2b, vp2b, {vp2a}, {}, *cdp);
+
+      Node vp3 = nm->mkNode(kind::SEXPR, d_cl, andNode.notNode(), children[0]);
+      success &= addAletheStep(
+          AletheRule::DUPLICATED_LITERALS, vp3, vp3, {vp2b}, {}, *cdp);
```

Comment:
All of this should be guarded by whether `n = 1`, no? Since in that case you can directly say that `vp3` is the same as `vp1`, and otherwise you do the computation that is happening here.

Reply:
I did that once and it seemed to make no difference, because the proof was not updated in that case.

Reply:
Done